learning_paths/api/v1/views.py

```diff
@@ -105,3 +109,45 @@ def get(self, request, learning_path_uuid):
         if serializer.is_valid():
             return Response(serializer.data, status=status.HTTP_200_OK)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class LearningPathViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    ViewSet for listing all learning paths and retrieving a specific learning path's details,
+    including steps and associated skills.
+    """
+
+    queryset = LearningPath.objects.all()
+    permission_classes = (IsAuthenticated,)
+    pagination_class = PageNumberPagination
+    lookup_field = "uuid"
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return LearningPathListSerializer
+        return LearningPathDetailSerializer
+
+
+class CertificateAvailabilityView(APIView):
```

Comment:
@pkulkark, this repository should not handle any logic related to the certificates. Instead, the certificates API (which does not yet exist) should be queried from the MFE. However, we didn't intend to include this part in the MVP.

Reply:
Dropping 9fce1d059458d71d.